### PR TITLE
Swap @bentley/reality-data-client with new @itwin/reality-data-client

### PR DESCRIPTION
**Pull request by MarcBedard8**
Update Display-test-app

### CODE REVIEW

File: core/frontend/src/DisplayStyleState.ts [hunk 1]
```diff
@@ -287,6 +287,15 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
     return undefined !== model && this.settings.contextRealityModels.delete(model);
   }
 
+  /** Detach the first [ContextRealityModel]($common) that matches the specified name and url.
+   * @see [DisplayStyleSettings.contextRealityModels]($common)
+   * @see [ContextRealityModels.delete]($common)
+   */
+  public detachRealityModelByKey(rdSourceKey: RealityDataSourceKey): boolean {
```
Comment (pmconne) [inline core/frontend/src/DisplayStyleState.ts:294]: These new methods look like they belong on DisplayStyleSettings, not DisplayStyleState.
(I know, there's a whole slew of reality-data-related methods on DisplayStyleState, but they're mostly not great APIs and don't belong there either).

File: test-apps/display-test-app/src/frontend/ClassificationsPanel.ts [hunk 2]
```diff
@@ -23,6 +23,16 @@ function clearElement(element: HTMLElement): void {
 
 const NO_MODEL_ID = "-1";
 
+enum RealityDataType {
```
Comment (pmconne) [inline test-apps/display-test-app/src/frontend/ClassificationsPanel.ts:26]: I think I've seen this enum before...did it move somewhere that display-test-app (and other apps) cannot access it? Is that why it's duplicated here?

File: core/frontend/src/DisplayStyleState.ts (hunk 1 above)
Comment (MarcBedard8) [reply core/frontend/src/DisplayStyleState.ts:295]: I marked them as internal for now so I can make DTA to work again. Not sure I want to keep them anyway.

Comment (pmconne) [reply core/frontend/src/DisplayStyleState.ts:295]: Adding "temporary" things to DisplayStyleState just to get something working is exactly how the display style reality model APIs became such a mess in the first place. If you need these new methods, please put them in the correct place.

File: test-apps/display-test-app/src/frontend/ClassificationsPanel.ts (hunk 2 above)
Comment (MarcBedard8) [reply test-apps/display-test-app/src/frontend/ClassificationsPanel.ts:26]: You probably see this in the old client. I'm not sure where I should put this. What do you think if we add an enum called SupportedRealityDataType in ContextRealityModel?

Comment (pmconne) [reply test-apps/display-test-app/src/frontend/ClassificationsPanel.ts:26]: Seems like the enum ought to be provided by the same package that provides ITwinRealityData - reality-data-client.

File: core/frontend/src/DisplayStyleState.ts (hunk 1 above)
Comment (MarcBedard8) [reply core/frontend/src/DisplayStyleState.ts]: Moved in Display-test-app

File: test-apps/display-test-app/src/frontend/ClassificationsPanel.ts (hunk 2 above)
Comment (MarcBedard8) [reply test-apps/display-test-app/src/frontend/ClassificationsPanel.ts]: Removed, will be added at the correct place

**Review by pmconne — APPROVED** [review]

**Comment by pmconne** [comment]
Queued an ImageTests run of the RealityTests test cases.

File: common/changes/@itwin/core-frontend/marc.bedard8-update_displayTestApp_2021-12-16-18-32.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/core-frontend",
+      "comment": "Swap old @bentley/reality-data-client with new @itwin/relaity-data-client, Update DisplayTestApp and add two new utility function and update API",
```
Comment (calebmshafer) [inline common/changes/@itwin/core-frontend/marc.bedard8-update_displayTestApp_2021-12-16-18-32.json:5]: Doesn't look like this changelog is needed anymore

**Review by calebmshafer — COMMENTED** [review]
Can you update the following files to remove the reality-data client from them?

- `.vscode/launch.json`
- `common/config/azre-pipelines/templates/integration-test-steps`
- `docs/reference/index.md`
- `docs/reference/leftNav.md`

File: common/changes/@itwin/core-frontend/marc.bedard8-update_displayTestApp_2021-12-16-18-32.json (hunk 3 above)
Comment (MarcBedard8) [reply common/changes/@itwin/core-frontend/marc.bedard8-update_displayTestApp_2021-12-16-18-32.json]: Removed

**Comment by MarcBedard8** [comment]
> Can you update the following files to remove the reality-data client from them?
> 
> * `.vscode/launch.json`
> * `common/config/azre-pipelines/templates/integration-test-steps`
> * `docs/reference/index.md`
> * `docs/reference/leftNav.md`

Nothing found in .vscode/launch.json, other references removed

**Review by calebmshafer — APPROVED** [review]
LGTM

**Comment by pmconne** [comment]
ImageTests completed successfully.

File: common/config/azure-pipelines/templates/integration-test-steps.yaml [hunk 4]
```diff
@@ -24,8 +24,6 @@ steps:
     displayName: rush install
 
   - script: |
-      node ./common/scripts/install-run-rush build -v --to reality-data-client
```
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/integration-test-steps.yaml:27]: You'll want to remove the lines 38-41 and 68-74

File: common/api/core-frontend.api.md [hunk 5]
```diff
@@ -5898,7 +5898,7 @@ export abstract class MapTilingScheme {
     // (undocumented)
     readonly numberOfLevelZeroTilesY: number;
     // (undocumented)
-    get rootLevel(): 0 | -1;
+    get rootLevel(): -1 | 0;
```
Comment (calebmshafer) [inline common/api/core-frontend.api.md:5901]: I'd revert this change, not sure why it flipped the order for you.

Comment (calebmshafer) [reply common/api/core-frontend.api.md]: pushed a change to revert it.

File: common/config/azure-pipelines/templates/integration-test-steps.yaml (hunk 4 above)
Comment (calebmshafer) [reply common/config/azure-pipelines/templates/integration-test-steps.yaml:27]: removed.

**Comment by calebmshafer** [comment]
@bsteinbk @NancyMcCallB mind taking a look when you get a chance?

**Comment by MarcBedard8** [comment]
@calebmshafer Not sure what is the problem with integration test. Getting this error: Error occurred in handler for 'itwin.electron.auth.getAccessToken': Authorization error: Not signed In. First call signIn()

**Comment by calebmshafer** [comment]
> @calebmshafer Not sure what is the problem with integration test. Getting this error: Error occurred in handler for 'itwin.electron.auth.getAccessToken': Authorization error: Not signed In. First call signIn()

@MarcBedard8 strange, it looks like the scopes are failing to be granted for the electron client. Let me see what's going on...

**Comment by calebmshafer** [comment]
@Merigifyio update

**Comment by calebmshafer** [comment]
@Mergifyio update

**Review by NancyMcCallB — APPROVED** [review]
